### ISSUE TITLE
index.html in project directory causes failure to build or load app on any device

### ISSUE THREAD

**Issue by kalzoo** [expo/expo#13114]
### Summary

First off, expo is awesome and thanks for all you do! Reporting this because it took quite a while to figure out since there was no information in the error message or anywhere else I could find.

If there's an `index.html` at the top level of an expo project, you'll get this error in the splash screen from any iOS simulator or real device. It may _also_ occur on other device types but I did not try.

> The data couldn't be read because it isn't in the correct format

![image](https://user-images.githubusercontent.com/22137047/120256819-30628c80-c24c-11eb-83a9-b86cc839b695.png)

The bundle doesn't even start to build, and there were no error messages anywhere I could find that tipped me off to what had happened. I was lucky enough to see and notice the file and that it didn't belong there, but if this had been anything less obvious it may have taken me much longer to bisect the issue and find the root cause. At the least, I'm reporting this issue so others might find it in search.

Ideally: the presence of a file like that shouldn't cause this failure.

At least: an error message _somewhere_ should make it more apparent what's happening.

Thanks!

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

41.0.1

### Environment

```
➜  native git:(master) ✗ expo diagnostics

  Expo CLI 4.5.1 environment info:
    System:
      OS: macOS 11.4
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.15.3 - /usr/local/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.9 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    IDEs:
      Xcode: 12.5/12E262 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^41.0.0 => 41.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-41.0.0.tar.gz => 0.63.2 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.5.1
    Expo Workflow: managed
```

### Reproducible demo or steps to reproduce from a blank project

```
expo init my-repro-project
cd my-repro-project
touch index.html
yarn ios
```

**Comment by brentvatne** [comment]
this looks to be a problem that we have inherited from react-native's metro server - notice if you initialize a project with `npx react-native init` and then add `index.html` in the root, then run `npm start` and navigate to `localhost:8081` in your browser you will see the contents of that `index.html` file. we use `/` endpoint to serve the expo manifest, and in the case of there being an `index.html` file metro server will serve that up rather than hitting our manifest handler. thanks for the report, we'll look into it

**Comment by kalzoo** [comment]
Thanks @brentvatne for the quick follow-up and fix! Makes perfect sense.

**Comment by brentvatne** [comment]
fixed in #3532 and will be released the next time we publish an update to expo-cli, thanks for the info!
